pkg/scalers/newrelic_scaler_test.go

```diff
@@ -0,0 +1,189 @@
+type newrelicAuthMetadataTestData struct {
+	metadata   map[string]string
+	authParams map[string]string
+	isError    bool
+}
+
+
+
+
+func TestPrometheusScalerAuthParams(t *testing.T) {
+	for _, testData := range testPrometheusAuthMetadata {
+		meta, err := parsePrometheusMetadata(&ScalerConfig{TriggerMetadata: testData.metadata, AuthParams: testData.authParams})
+
+		if err != nil && !testData.isError {
+			t.Error("Expected success but got error", err)
+		}
+		if testData.isError && err == nil {
+			t.Error("Expected error but got success")
+		}
+
+		if err == nil {
+			if (meta.enableBearerAuth && !strings.Contains(testData.metadata["authModes"], "bearer")) ||
+				(meta.enableBasicAuth && !strings.Contains(testData.metadata["authModes"], "basic")) ||
+				(meta.enableTLS && !strings.Contains(testData.metadata["authModes"], "tls")) {
+				t.Error("wrong auth mode detected")
+			}
+		}
+	}
+}
+
+type prometheusQromQueryResultTestData struct {
+	name           string
+	bodyStr        string
+	responseStatus int
+	expectedValue  float64
+	isError        bool
+}
+
+var testPromQueryResult = []prometheusQromQueryResultTestData{
+	{
+		name:           "no results",
+		bodyStr:        `{}`,
+		responseStatus: http.StatusOK,
+		expectedValue:  0,
+		isError:        false,
+	},
+	{
+		name:           "no values",
+		bodyStr:        `{"data":{"result":[]}}`,
+		responseStatus: http.StatusOK,
+		expectedValue:  0,
+		isError:        false,
+	},
+	{
+		name:           "valid value",
+		bodyStr:        `{"data":{"result":[{"value": ["1", "2"]}]}}`,
+		responseStatus: http.StatusOK,
+		expectedValue:  2,
+		isError:        false,
+	},
+	{
+		name:           "not enough values",
+		bodyStr:        `{"data":{"result":[{"value": ["1"]}]}}`,
+		responseStatus: http.StatusOK,
+		expectedValue:  -1,
+		isError:        true,
+	},
+	{
+		name:           "multiple results",
+		bodyStr:        `{"data":{"result":[{},{}]}}`,
+		responseStatus: http.StatusOK,
+		expectedValue:  -1,
+		isError:        true,
+	},
+	{
+		name:           "error status response",
+		bodyStr:        `{}`,
+		responseStatus: http.StatusBadRequest,
+		expectedValue:  -1,
+		isError:        true,
+	},
+}
+
+func TestPrometheusScalerExecutePromQuery(t *testing.T) {
+	for _, testData := range testPromQueryResult {
+		t.Run(testData.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+				writer.WriteHeader(testData.responseStatus)
+
+				if _, err := writer.Write([]byte(testData.bodyStr)); err != nil {
+					t.Fatal(err)
+				}
+			}))
+
+			scaler := prometheusScaler{
+				metadata: &prometheusMetadata{
+					serverAddress: server.URL,
+				},
+				httpClient: http.DefaultClient,
+			}
+
+			value, err := scaler.ExecutePromQuery(context.TODO())
+
+			assert.Equal(t, testData.expectedValue, value)
+
+			if testData.isError {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+*/
```

Comment:
Are these lines pending changes or they could be deleted?

Reply:
Yes I will delete this, I'm out of the office until Dec 6,will make those changes on that date, hopefully early if I can get access to my laptop without the wife killing me :)

Reply:
No worries, no need to rush :)